### PR TITLE
Rianfowler/sc 59400/add typechecks to dashboard child components dashboard version card

### PR DESCRIPTION
**Pull request by rianfowler**
#### What this PR does / why we need it:
- adds typechecking to dashboard version card

#### Which issue(s) this PR fixes:
https://app.shortcut.com/replicated/story/59400/add-typechecks-to-dashboard-child-components

#### Does this PR introduce a user-facing change?
none -refactor 
#### Does this PR require documentation?
none

### CODE REVIEW

File: web/src/types/index.ts [hunk 1]
```diff
@@ -88,12 +95,63 @@ export type DashboardActionLink = {
...
+  preflightSkipped: boolean;
+  releaseNotes: string;
   semver: string;
   sequence: number;
+  source: string;
   versionLabel?: string;
+  status: VersionStatus;
+  title: string;
+  yamlErrors: string[];
 };
```
Comment (rianfowler) [inline web/src/types/index.ts:134]: this type is so big haha

**Review by rianfowler — COMMENTED** [review]
pretty straight forward- no weird stuff in this one

File: web/src/components/Icon.tsx [hunk 2]
```diff
@@ -7,11 +7,11 @@ import "@src/css/icon.css";
...
   size: number | string;
-  color: string;
-  style: object;
-  className: string;
-  disableFill: boolean;
-  removeInlineStyle: boolean;
+  color?: string;
+  style?: object;
+  className?: string;
+  disableFill?: boolean;
```
Comment (rianfowler) [inline web/src/components/Icon.tsx:13]: made some of these optional

File: web/src/components/Icon.tsx [hunk 3]
```diff
@@ -7,11 +7,11 @@ import "@src/css/icon.css";
...
+  removeInlineStyle?: boolean;
```
Comment (miaawong) [inline web/src/components/Icon.tsx:14]: thanks for doing that!

**Review by miaawong — APPROVED** [review]

**Pull request merged**